docs/deployment/getting-started.md

```diff
@@ -0,0 +1,180 @@
+                                  populate remote address info.
+  --server-header / --no-server-header
+                                  Enable/Disable default Server header.
+  --date-header / --no-date-header
+                                  Enable/Disable default Date header.
+  --forwarded-allow-ips TEXT      Comma separated list of IPs to trust with
+                                  proxy headers. Defaults to the
+                                  $FORWARDED_ALLOW_IPS environment variable if
+                                  available, or '127.0.0.1'.
+  --root-path TEXT                Set the ASGI 'root_path' for applications
+                                  submounted below a given URL path.
+  --limit-concurrency INTEGER     Maximum number of concurrent connections or
+                                  tasks to allow, before issuing HTTP 503
+                                  responses.
+  --backlog INTEGER               Maximum number of connections to hold in
+                                  backlog
+  --limit-max-requests INTEGER    Maximum number of requests to service before
+                                  terminating the process.
+  --timeout-keep-alive INTEGER    Close Keep-Alive connections if no new data
+                                  is received within this timeout.  [default:
+                                  5]
+  --ssl-keyfile TEXT              SSL key file
+  --ssl-certfile TEXT             SSL certificate file
+  --ssl-keyfile-password TEXT     SSL keyfile password
+  --ssl-version INTEGER           SSL version to use (see stdlib ssl module's)
+                                  [default: 17]
+  --ssl-cert-reqs INTEGER         Whether client certificate is required (see
+                                  stdlib ssl module's)  [default: 0]
+  --ssl-ca-certs TEXT             CA certificates file
+  --ssl-ciphers TEXT              Ciphers to use (see stdlib ssl module's)
+                                  [default: TLSv1]
+  --header TEXT                   Specify custom default HTTP response headers
+                                  as a Name:Value pair
+  --version                       Display the uvicorn version and exit.
+  --app-dir TEXT                  Look for APP in the specified directory, by
+                                  adding this to the PYTHONPATH. Defaults to
+                                  the current working directory.  [default: .]
+  --h11-max-incomplete-event-size INTEGER
+                                  For h11, the maximum number of bytes to
+                                  buffer of an incomplete event.
+  --factory                       Treat APP as an application factory, i.e. a
+                                  () -> <ASGI app> callable.
+  --help                          Show this message and exit.
+```
+
+
+See the [settings documentation](../settings.md) for more details on the supported options for running uvicorn.
+
+### Running programmatically
```

Comment:
I think we should also give an example of using `uvicorn.Config` + `uvicorn.Server` manually. I know that @Tinche and I think @agronholm  (sorry if I am misremembering) use this style a lot . It lets you take control of the event loop as well as startup/shutdown, configuring logging and several other things.

Reply:
I haven’t tried that implementation yet. I’ll add that too!